src/Promitor.Tests.Unit/Validation/Scraper/Metrics/AzureMetricConfigurationValidatorTest.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using Promitor.Agents.Scraper.Validation.MetricDefinitions;
+using Promitor.Core.Scraping.Configuration.Model;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
+using Xunit;
+using MetricDimension = Promitor.Core.Scraping.Configuration.Model.MetricDimension;
+
+namespace Promitor.Tests.Unit.Validation.Scraper.Metrics;
+
+public class AzureMetricConfigurationValidatorTest
+{
+    [Fact]
+    public void DimensionsAndDimension_Fails()
+    {
+        // Arrange
+        var metricConfig = new AzureMetricConfiguration
+        {
+            MetricName = "testMetric",
+        };
+
+        var metricDefinition = new MetricDefinition
+        {
+            AzureMetricConfiguration = metricConfig
+        };
+
+        metricConfig.Dimensions = new List<MetricDimension> { new(), new() };
```

Comment:
Nit - WHy not add to initialization on line 18?

Reply:
Honestly just because I hadn't used the expression in line 16 before, so I didn't think to include those two there too 😄
Added it in all cases now.